src/middlewared/middlewared/api/v25_10_0/nvmet_global.py

```diff
@@ -0,0 +1,95 @@
+from middlewared.api.base import BaseModel, Excluded, ForUpdateMetaclass, excluded_field, single_argument_args
+from .common import QueryFilters, QueryOptions
+
+__all__ = [
+    "NVMetGlobalEntry",
+    "NVMetGlobalUpdateArgs",
+    "NVMetGlobalUpdateResult",
+    "NVMetGlobalAnaEnabledArgs",
+    "NVMetGlobalAnaEnabledResult",
+    "NVMetGlobalRDMAEnabledArgs",
+    "NVMetGlobalRDMAEnabledResult",
+    "NVMetGlobalSessionsArgs",
+    "NVMetGlobalSessionsResult",
+]
+
+
+class NVMetGlobalEntry(BaseModel):
+    id: int
+    basenqn: str
+    """
+    NQN to be used as the prefix on the creation of a subsystem, if a subnqn is not supplied to `nvmet.subsys.create`.
+
+    Modifying this value will *not* change the subnqn of any existing subsystems.
+    """
+    kernel: bool = True
```

Comment:
What do these default values mean?

Reply:
An alternative (non-kernel) backend is discussed in NEP-058.  Plumbing here to facilitate.

Reply:
Where will this default value be used?

Reply:
At the moment it is used in e.g. `load_kernel_modules` and some `namespace` private APIs (e.g. `resize_namespace`).

It is anticipated that it'll be used more later.

Reply:
@themylogin 
> Am I missing something?

As usual, no!  I didn't grasp that you were talking **specifically** about the default values are shown in the API file.

I was responding in more general terms, e.g. the global `kernel` setting.

Which gets a default value from global.py
```
    nvmet_global_basenqn = sa.Column(sa.String(255), default=uuid_nqn)
    nvmet_global_kernel = sa.Column(sa.Boolean(), default=True)
    nvmet_global_ana = sa.Column(sa.Boolean(), default=False)
    nvmet_global_rdma = sa.Column(sa.Boolean(), default=False)
    nvmet_global_xport_referral = sa.Column(sa.Boolean(), default=True)
```

So, I suspect you are correct.  These default values can be removed from the API.  I'll remove / build / retest.  Thanks.